app/src/main/java/com/iti/android/team1/ebuy/ui/login_screen/viewmodel/LoginScreenViewModel.kt

```diff
@@ -13,23 +13,46 @@ import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
 
+
 class LoginScreenViewModel(private val repository: IRepository) : ViewModel() {
 
     private val _loginState: MutableStateFlow<ResultState<Customer>> =
         MutableStateFlow(ResultState.Loading)
     val loginState get() = _loginState.asStateFlow()
 
-    fun makeLoginRequest(customerLogin: CustomerLogin) {
+    fun makeLoginRequest(email: String, password: String) {
+
+        val customerLogin = CustomerLogin(
+            email = email,
+            password = repository.encodePassword(password)
+        )
+
         viewModelScope.launch(Dispatchers.IO) {
             val result = async { repository.loginCustomer(customerLogin) }
-            setLoginState(result.await())
+            setLoginState(result.await(), password)
         }
     }
 
-    private fun setLoginState(result: NetworkResponse<Customer>) {
+    fun setUserIdToPrefs(userId: Long) = repository.setUserIdToPrefs(userId)
+
+    fun setAuthStateToPrefs(state: Boolean) = repository.setAuthStateToPrefs(state)
+
+    fun getAuthStateFromPrefs() = repository.getAuthStateFromPrefs()
+
+    private suspend fun setLoginState(result: NetworkResponse<Customer>, password: String) {
         when (result) {
-            is NetworkResponse.FailureResponse -> ResultState.Error(result.errorString)
-            is NetworkResponse.SuccessResponse -> ResultState.Success(result.data)
+            is NetworkResponse.FailureResponse -> {
+                _loginState.emit(ResultState.Error(result.errorString))
+            }
+            is NetworkResponse.SuccessResponse -> {
+                if (password == repository.decodePassword(result.data.password
+                        ?: "")
+                )
+                    _loginState.emit(ResultState.Success(result.data))
+                else
+                    _loginState.emit(ResultState.Error("Invalid data"))
```

Comment:
why do you check for password here?

Reply:
the API was checking on the email only not on the password so i that's was the only way to make that this is the user, ill updated after we managed to filter the data

Reply:
The API returns a response if the email and password are the same as in the API, thus you don't need to check for the password here if the answer is success, but you will emit an error if there is no customer in the response.